src/client/scripts/esm/chess/logic/movepiece.js

```diff
@@ -452,123 +308,84 @@ function rewindGameToIndex(gamefile, moveIndex, { removeMove = true, updateData
-
-	if (move.captured) { // Replace the piece captured
-		const type = move.captured;
-		addPiece(gamefile, type, move.endCoords, move.rewindInfo.capturedIndex, { updateData });
-	}
-
-	if (animate) animation.animatePiece(move.type, move.endCoords, move.startCoords);
+	updateTurn(gamefile);
 }
 
 /**
- * Simulates the provided move, testing if it's in check and, if specified, also the game conclusion,
- * then undo's the move, restoring it to how the gamefile was before.
- * @param {gamefile} gamefile - The gamefile
- * @param {Move} The move to simulate.
- * @param {string} colorToTestInCheck - The side to test if they are in check. Usually this is the color of the side making the move, because we don't want to step into check.
- * @param {Object} options - An object that may contain the properties:
- * - `doGameOverChecks`: Whether, while simulating this move, to perform game over checks such as checkmate or other win conditions. SLOWER, but this can be used to verify the game conclusion the opponent claimed. Default: *false*
- * @returns {Object} An object that may contains the properties:
- * - `isCheck`: Whether making this move puts the specified color in check. Usually it's you stepping into check.
- * - `gameConclusion`: The resulting `gameConclusion` after the move, if `doGameOverChecks` was specified as *true*.
+ * Wraps a function in a simulated move
+ * @param {gamefile} gamefile 
+ * @param {Move} move 
+ * @param {CallableFunction} callback
+ * @returns whatever is returned by the callback
  */
-function simulateMove(gamefile, move, colorToTestInCheck, { doGameOverChecks = false } = {}) {
+function simulateMoveWrapper(gamefile, move, callback) {
 	// Moves the piece without unselecting it or regenerating the pieces model.
-	makeMove(gamefile, move, { pushClock: false, animate: false, updateData: false, simulated: true, doGameOverChecks, updateProperties: doGameOverChecks });
-    
+	generateMove(gamefile, move);
+	makeMove(gamefile, move);
+
 	// What info can we pull from the game after simulating this move?
-	const info = {
-		isCheck: doGameOverChecks ? gamefile.inCheck : checkdetection.detectCheck(gamefile, colorToTestInCheck, []),
-		gameConclusion: doGameOverChecks ? gamefile.gameConclusion : undefined
-	};
+	const info = callback();
 
 	// Undo the move, REWIND.
 	// We don't have to worry about the index changing, it is the same.
 	// BUT THE CAPTURED PIECE MUST be inserted in the exact location!
 	// Only remove the move
-	rewindMove(gamefile, { updateData: false, animate: false });
+	rewindMove(gamefile, true);
+
+	return info;
+}
 
-	return info; // Info from simulating the move: { isCheck, gameConclusion }
+/**
+ * Simulates a move to get the check
+ * @param {gamefile} gamefile 
+ * @param {Move} move 
+ * @param {*} colorToTestInCheck 
+ * @returns 
+ */
+function getSimulatedCheck(gamefile, move, colorToTestInCheck) {
+	return simulateMoveWrapper(
```

Comment:
It doesn't look like the simulated move wrapper knows when to perform game over checks or not, it always does by default. This is inefficient to run the checkmate algorithm when all we want to know is if the position is in check

Reply:
Wdym?

Reply:
`movepiece.makeMove` doesn't update conclusion anymore

Reply:
Oh okay, there's 2 makeMove()'s